src/plugins/pcr_primer_design_for_dna_assembly/src/PCRPrimerDesignForDNAAssemblyOPWidget.cpp

```diff
@@ -0,0 +1,236 @@
+    }
+    connect(tbLoadBackbone, &QAbstractButton::clicked, this, &PCRPrimerDesignForDNAAssemblyOPWidget::sl_loadBackbone);
+    connect(tbsaveRandomSequences, &QAbstractButton::clicked, this, &PCRPrimerDesignForDNAAssemblyOPWidget::sl_saveRandomSequences);
+    connect(tbLoadOtherSequencesInPcr, &QAbstractButton::clicked, this, &PCRPrimerDesignForDNAAssemblyOPWidget::sl_loadOtherSequenceInPcr);
+}
+
+void PCRPrimerDesignForDNAAssemblyOPWidget::sl_start() {
+    //TODO
+}
+
+void PCRPrimerDesignForDNAAssemblyOPWidget::sl_selectManually() {
+    auto currentSMButton = qobject_cast<QToolButton*>(sender());
+    SAFE_POINT(currentSMButton != nullptr, L10N::nullPointerError("QAbstractButton"), );
+
+    if (smButton != nullptr) {
+        bool disableState = true;
+        if (smButton->isChecked()) {
+            smButton->setChecked(false);
+            disableState = false;
+        }
+        smButton = nullptr;
+        start = nullptr;
+        end = nullptr;
+        disconnect(updateRegionConnection);
+        updateRegionConnection = QMetaObject::Connection();
+
+        if (disableState) {
+            return;
+        }
+    }
+
+    smButton = currentSMButton;
+    if (smButton == tbLeftAreaSelectManually) {
+        start = sbLeftAreaStart;
+        end = sbLeftAreaEnd;
+    } else {
+        start = sbRightAreaStart;
+        end = sbRightAreaEnd;
+    }
+    auto activeSequenceContext = annDnaView->getActiveSequenceContext();
+    SAFE_POINT(activeSequenceContext != nullptr, L10N::nullPointerError("ADVSequenceObjectContext"), );
+
+    auto sequenceSelection = activeSequenceContext->getSequenceSelection();
+    SAFE_POINT(sequenceSelection != nullptr, L10N::nullPointerError("DNASequenceSelection"), );
+
+    updateRegionConnection = connect(sequenceSelection, &DNASequenceSelection::si_onSelectionChanged, this, &PCRPrimerDesignForDNAAssemblyOPWidget::sl_updateRegion);
+}
+
+void PCRPrimerDesignForDNAAssemblyOPWidget::sl_updateRegion() {
+    auto sequenceSelection = qobject_cast<DNASequenceSelection*>(sender());
+    SAFE_POINT(sequenceSelection != nullptr, L10N::nullPointerError("DNASequenceSelection"), );
+
+    const auto& selectedRegions = sequenceSelection->getSelectedRegions();
+    CHECK(selectedRegions.size() != 0, );
+
+    if (selectedRegions.size() > 1) {
+        coreLog.details(tr("Multiple regions selection, the only first is set as primer search area."));
+    }
+    const auto& region = selectedRegions.first();
+    start->setValue(region.startPos + 1);
+    end->setValue(region.endPos());
+}
+
+void PCRPrimerDesignForDNAAssemblyOPWidget::sl_updateMaxValues() {
+    auto activeSequenceContext = annDnaView->getActiveSequenceContext();
+    SAFE_POINT(activeSequenceContext != nullptr, L10N::nullPointerError("ADVSequenceObjectContext"), );
+
+    auto seqLength = activeSequenceContext->getSequenceLength();
+    for (const auto& minSb : minMaxValuesSb.keys()) {
```

Comment:
qAsConst

Reply:
error C2280: 'void qAsConst<QList<Key>>(const T &&)': attempting to reference a deleted function

Reply:
There is a reason for this error: you need to cache keys as a local var to avoid this error - this is a QT recommendation (and the reason why qAsConst exists)

Reply:
This is a very vague comment - please, detail what you want from me

Reply:
Please check this bug for details: https://ugene.dev/tracker/browse/UGENE-6992.
Long story short: for-loop and QT collections in UGENE should be used with qAsConst.

Reply:
Fixed